kolena/_experimental/dataset/_dataset.py

```diff
@@ -113,15 +121,52 @@ def _to_deserialized_dataframe(df: pd.DataFrame, column: str) -> pd.DataFrame:
     return result
 
 
-def register_dataset(name: str, df: pd.DataFrame) -> None:
+def _upload_dataset_chunk(df: pd.DataFrame, load_uuid: str, id_fields: List[str]) -> None:
```

Comment:
We're doing chunking inside `upload_data_frame`. Is there reason for 2-level of chunk control?

Reply:
Chunking here allows the user to send their dataframe to upload in chunks, and the bach_load chunking makes sure we are not upload a giant file at once.
There might not be a need for it in dataset as the chunks size on the dataset should always be smaller than batch_load chunk size.  However, for other things that does not have chunk support in the respective upload function, we still need the batch_load chunking